### PR TITLE
My VA Health user provisioning (ToU cookie)

### PR DESCRIPTION
**Pull request by asg5704**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary
This PR includes the body of work necessary for the Cerner (My VA Health) cookie as it pertains to Terms of Use. It essentially does the following:
- [x] Creates a new route for Cerner provisioning on `/terms-of-use/myvahealth`
- [x] Creates a component for the above route that attempts auto-provisions My VA Health users
- [x] Adds unit tests for the component

## Related issue(s)

- department-of-veterans-affairs/va.gov-team#69981

## Testing done
- Unit testing + manual

## Screenshots
n/a - While this IS a component, no UI other than a va-loading-indicator is shown to the user.

## What areas of the site does it impact?
This should only impact My VA Health (Cerner) users in staging & below

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/terms-of-use/components/MyVAHealth.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        apiRequest(`/terms_of_use_agreements/update_provisioning`, {
+          method: 'PUT',
+          headers: { 'Content-Type': 'application/json' },
+          credentials: 'include',
+        })
+          .then(response => {
+            if (response?.provisioned) {
+              window.location = parseRedirectUrl(
+                decodeURIComponent(ssoeTarget),
+              );
```
[the remark this answers is not in the document]
Comment (joeniquette) [reply src/applications/terms-of-use/components/MyVAHealth.jsx:33]: @asg5704 can you provide justification for the false positive for documentation purposes please. Do we sanitize this input before processing for example?

[the remark this answers is not in the document]
Comment (asg5704) [reply src/applications/terms-of-use/components/MyVAHealth.jsx:33]: Yes, we sanitize the input in the `parseRedirectUrl` function similar to how we do so in the normal ToU. We abstract out the domain name and verify it against what we expect.

File: src/applications/terms-of-use/tests/MyVAHealth.unit.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import { render, waitFor } from '@testing-library/react';
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+
+import { $ } from 'platform/forms-system/src/js/utilities/ui';
+import MyVAHealth from '../components/MyVAHealth';
+
+const oldLocation = global.window.location;
+
+Array.from({ length: 200 }).forEach(() => {
```
Comment (afialydia) [inline src/applications/terms-of-use/tests/MyVAHealth.unit.spec.jsx:13]: Are we running the test this many times to simulate continuous use/ strain on the route?

Comment (asg5704) [reply src/applications/terms-of-use/tests/MyVAHealth.unit.spec.jsx:13]: Oh weird I thought I took that out. Good catch Afia

**Review by acaruso-oddball — COMMENTED** [review]
looks good overall, the logic makes sense.

**Review by acaruso-oddball — APPROVED** [review]

**Pull request merged**